### DISCUSSION TITLE
Recommendations for Magento2 Dev Team local environments on MacBook Pro M2 w/ 96 RAM, and Skilling up to M2 best practices

### DISCUSSION THREAD

**Discussion by beseiso** [wardenenv/warden#615] [Help & Support] [answered]
My team and I are receiving MacBook Pros with M2 chips with max RAM.

We have the option of direct install, Docker, Warden, etc.

Our intention is to enable Devs to have multiple environments simultaneously, that way, they can support production bug fixes by having mirror of Prod code, new features in a Dev instance, and maybe a sandbox to Tinker/Innovate in, etc.

What recommendations does the Magento community have for local development environments: Direct install, Docker, or Warden+Docker?

What recommendations for IDE? Visual Studio Code, Visual Studio, PHP Storm, Sublime, etc.?

What recommendations to skill up the Magento Developers in streamlined manner from Novice to Best Practices?

In advance, appreciate Experts who are seasoned Magento Developers, to provide insights.

Thank you,
JSB

**Comment by navarr** [comment, marked as answer]
Well, you're asking in the Warden community what their recommendation is.

Warden is good for solo sites but is especially useful for working on multiple client sites with differing requirements. By default for Mac, it will setup a DNS server, allow certificate signing so you're testing on HTTPS, and keep databases completely separate.

You can turn each site on or off to maintain CPU/RAM.  

As far as IDEs, that's definitely down to preference. The two most popular are JetBrains PhpStorm and VSCode. PhpStorm is well positioned as it's been popular longer and has an official Magento extension.

For Training, there's always our sponsor SwiftOtter.

**Reply by beseiso** [reply]
@navarr Appreciate these insights. So, Warden works well with the Apple M2 silicone?

I'll suggest/recommend your guidance on PhpStorm to our Magento Dev Team.
I'm the Project Manager, yet wanting to skill up as well. So, I'll be setting up my new Apple MacBook Pro with M2 chip with Warden. 
I finished the Warden install on my 2012 MacBook Pro that uses Intel chipset and it works well. Didn't know if the Apple M2 chip works. I'm yet to receive my new laptop and want to ensure Apple M2 chips/Architecture is supported.

As for training, I just registered with SwiftOtter training. Thank you for the suggestion, and support.

Appreciate you!

Wonderful weekend to you, and to all @navarr!

Thank you!

**Reply by navarr** [reply]
In fairness Apple silicon is not yet super well supported. Most services run on Apple silicon but at current PHP still needs Rosetta translating x86 to Apple. 

That will result in a performance decrease, but once we've got it working it'll be an easy, painless upgrade.

We've made recent strides with wardenenv/images#4 but it's not quite ready

**Comment by dudzio12** [comment]
As for now I could recommend den instead of warden, since it support Apple Silicon.
For IDE -PHPStorm and Sublime for occasional reading of large files.
TablePlus is great tool for db management.
Sublime Merge for tracking repository changes. It's much faster than Source Tree.
